jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -197,6 +198,58 @@ def __init__(self, *args, **kwargs):
         # default_tool_priority=['jdaviz:selectslice']
         super().__init__(*args, **kwargs)
 
+    def _get_spatial_subset_layers(self):
+        return [ls for ls in self.state.layers
+                if isinstance(getattr(ls.layer, 'subset_state', None), RoiSubsetState)]
+
+    def _get_spectral_subset_layers(self):
+        return [ls for ls in self.state.layers
+                if isinstance(getattr(ls.layer, 'subset_state', None), RangeSubsetState)]
```

Comment:
Can't we combine this into one and swap the class to be compared depending on a keyword?